pkg/storage/memory/memory.go

```diff
@@ -120,14 +124,17 @@ type MemoryBackend struct {
 	assertions map[string][]*openfgav1.Assertion
```

Comment:
The fields`changes`, `stores`, `assertions` should also have `// GUARDED_BY(mu).`

The mutex is used everywhere 😆 This is the subject of https://github.com/openfga/openfga/issues/1138